api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -14,14 +14,18 @@ class CartController < ::Spree::Api::V2::BaseController
           def create
             spree_authorize! :create, Spree::Order
 
-            order_params = {
+            create_cart_params = {
               user: spree_current_user,
               store: current_store,
-              currency: current_currency
+              currency: current_currency,
+              order_params: {
+                public_metadata: metadata_params[:public_metadata].to_h,
+                private_metadata: metadata_params[:private_metadata].to_h,
+              }
```

Comment:
Why do we need to nest metadata in `order_params`?

Reply:
We do not, but it keeps the params list shorter. Should I change it to not be nested?

Reply:
@aleksandarpetrushev this makes it inconsistent because the params here https://github.com/spree/spree/pull/11499/files#diff-ae18d7948c7abcd3c987e056883166574c833d2889994e46614e5c8f73d2ab02R37 are not nested, we should stick into one way of doing this - in this case, top-level, not nested